### PR TITLE
Default project values for testing

### PR DESCRIPTION
**Pull request by sarahmarshy**

### CODE REVIEW

File: project_generator/project.py [hunk 1]
```diff
@@ -197,21 +193,37 @@ def __init__(self, name, project_files, pgen_workspace):
...
+            'output_dir': {
+                'path': '',
+                'rel_path': '',
+                'rel_count': '',
+            },
+            'target': '',  # target
+            'template' : '',        # tool template
+            'output_type': self.output_types['executable'],   # output type, default to exe
+            'tools_supported': [self.workspace.settings.DEFAULT_TOOL]
+
```
Comment (0xc0170) [inline project_generator/project.py]: How do you like this, now it's clear entire dict what we work with

**Comment by 0xc0170** [comment]
Please rebase to the latest dev_v0.7

**Comment by 0xc0170** [comment]
I did not set target in the single_project.py which is used for exporting. Means we should have one target which supports all tools, to be able to test them. Looking at definitions, I would say mbed-lpc1768 is the one supporting them all. Then we should be able to bugfix those errors in Travis.

File: project_generator/project.py [hunk 2]
```diff
@@ -197,21 +193,37 @@ def __init__(self, name, project_files, pgen_workspace):
...
+            'core': '',                 # core
+            'linker_file': None,          # linker command file
+            'build_dir' : 'build',
+            'debugger' : 'cmsis-dap',   # TODO: find out what debugger is connected
+            'includes': [],             # include paths
+            'source_paths': [],         # source paths
+            'source_files_c': [],       # c source files
+            'source_files_cpp': [],     # c++ source files
+            'source_files_s': [],       # assembly source files
+            'source_files_obj': [{}],   # object files
```
Comment (0xc0170) [inline project_generator/project.py]: Why these are dict inside list? I recall I had to fix some errors using this for valinor, can't recall what was the error and why its still there.

File: project_generator/project.py [hunk 3]
```diff
@@ -197,21 +193,37 @@ def __init__(self, name, project_files, pgen_workspace):
...
+            'source_files_lib': [{}],   # libraries
+            'macros': [],               # macros (defines)
+            'misc': {},
+            'project_dir': {
+                'name': '.' + os.path.sep,
```
Comment (0xc0170) [inline project_generator/project.py]: shouldn't name be empty by default?

File: project_generator/project.py (hunk 2 above)
Comment (sarahmarshy) [inline project_generator/project.py]: Shouldn't be that way once flushed out in customize function. But, this is what it looked like in Valinor.

Comment (0xc0170) [inline project_generator/project.py]: Yes, I recall I received error if I sent only list :-)

**Pull request merged**